spot_wrapper/wrapper.py

```diff
@@ -739,47 +1227,48 @@ def trajectory_cmd(self, goal_x, goal_y, goal_heading, cmd_duration, frame_name=
-                self._robot_state_client.get_robot_state().kinematic_state.transforms_snapshot)
-            body_tform_goal = math_helpers.SE3Pose(x=goal_x, y=goal_y, z=0,
-                                                   rot=math_helpers.Quat.from_yaw(goal_heading))
+                self._robot_state_client.get_robot_state().kinematic_state.transforms_snapshot
+            )
+            body_tform_goal = math_helpers.SE3Pose(
+                x=goal_x, y=goal_y, z=0, rot=math_helpers.Quat.from_yaw(goal_heading)
+            )
             vision_tform_goal = vision_tform_body * body_tform_goal
             response = self._robot_command(
                 RobotCommandBuilder.synchro_se2_trajectory_point_command(
                     goal_x=vision_tform_goal.x,
                     goal_y=vision_tform_goal.y,
                     goal_heading=vision_tform_goal.rot.to_yaw(),
                     frame_name=frame_helpers.VISION_FRAME_NAME,
-                    params=mobility_params),
-                end_time_secs=end_time
+                    params=mobility_params,
+                ),
+                end_time_secs=end_time,
             )
-        elif frame_name == 'odom':
+        elif frame_name == "odom":
             odom_tform_body = frame_helpers.get_odom_tform_body(
-                self._robot_state_client.get_robot_state().kinematic_state.transforms_snapshot)
-            body_tform_goal = math_helpers.SE3Pose(x=goal_x, y=goal_y, z=0,
-                                                   rot=math_helpers.Quat.from_yaw(goal_heading))
+                self._robot_state_client.get_robot_state().kinematic_state.transforms_snapshot
+            )
+            body_tform_goal = math_helpers.SE3Pose(
+                x=goal_x, y=goal_y, z=0, rot=math_helpers.Quat.from_yaw(goal_heading)
+            )
             odom_tform_goal = odom_tform_body * body_tform_goal
             response = self._robot_command(
                 RobotCommandBuilder.synchro_se2_trajectory_point_command(
                     goal_x=odom_tform_goal.x,
                     goal_y=odom_tform_goal.y,
                     goal_heading=odom_tform_goal.rot.to_yaw(),
                     frame_name=frame_helpers.ODOM_FRAME_NAME,
-                    params=mobility_params),
-                end_time_secs=end_time
+                    params=mobility_params,
+                ),
+                end_time_secs=end_time,
             )
         else:
-            raise ValueError('frame_name must be \'vision\' or \'odom\'')
+            raise ValueError("frame_name must be 'vision' or 'odom'")
         if response[0]:
             self._last_trajectory_command = response[2]
         return response[0], response[1]
 
-    def robot_command(self, robot_command):
-        end_time = time.time() + MAX_COMMAND_DURATION
-        return self._robot_command(robot_command, end_time_secs=end_time,
-                                   timesync_endpoint=self._robot.time_sync.endpoint)
```

Comment:
This function should be restored please - we use it extensively!

Reply:
I removed this because I couldn't find any calls to this function within the wrapper or the ros2 driver code. Perhaps I missed something. Or do you mean that you are calling this from code that is external to the ros2/wrapper code?

Reply:
It's on line 636 of `spot_ros2.py` on the main branch of the ROS2 code in `handle_robot_command`.
```
    def handle_robot_command(self, goal_handle):
        ros_command = goal_handle.request.command
        proto_command = robot_command_pb2.RobotCommand()
        conv.convert_bosdyn_msgs_robot_command_to_proto(ros_command, proto_command)
        self._wait_for_goal = None
        if not self.spot_wrapper:
            self._wait_for_goal = WaitForGoal(self.node.get_clock(), 2.0)
            goal_id = None
        else:
            success, err_msg, goal_id = self.spot_wrapper.robot_command(proto_command)
            if not success:
                raise Exception(err_msg)
```

Admittedly that's the only call to it - when I said "use it extensively" I mean we use the robot command action a lot not that this is called from multiple places :)